File: app/src/main/java/com/vwoom/timelapsegallery/database/dao/TagDao.java
package com.vwoom.timelapsegallery.database.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.vwoom.timelapsegallery.database.entry.TagEntry;

import java.util.List;

@Dao
public interface TagDao {

    @Query("SELECT * FROM tag")
    List<TagEntry> loadAllTags();

    @Insert
    void insertTag(TagEntry tagEntry);

    @Delete
    void deleteTag(TagEntry tagEntry);

    @Update
    void updateTag(TagEntry tagEntry);

}

File: app/src/main/java/com/vwoom/timelapsegallery/database/dao/TagDao.kt
package com.vwoom.timelapsegallery.database.dao

import androidx.room.*
import com.vwoom.timelapsegallery.database.entry.TagEntry

@Dao
interface TagDao {
    @Query("SELECT * FROM tag")
    fun loadAllTags(): List<TagEntry?>?

    @Insert
    fun insertTag(tagEntry: TagEntry?)

    @Delete
    fun deleteTag(tagEntry: TagEntry?)

    @Update
    fun updateTag(tagEntry: TagEntry?)
}